Source/Scene/BufferLoader.js

```diff
@@ -98,32 +97,39 @@ Object.defineProperties(BufferLoader.prototype, {
  */
 BufferLoader.prototype.load = function () {
   if (defined(this._typedArray)) {
-    this._promise.resolve(this);
-    return;
+    this._promise = Promise.resolve(this);
+  } else {
+    this._promise = loadExternalBuffer(this);
   }
-
-  loadExternalBuffer(this);
+  return this._promise;
 };
 
 function loadExternalBuffer(bufferLoader) {
   const resource = bufferLoader._resource;
   bufferLoader._state = ResourceLoaderState.LOADING;
-  BufferLoader._fetchArrayBuffer(resource)
+  const promise = BufferLoader._fetchArrayBuffer(resource);
+
+  // TODO: I don't think this is OK-- The request is cancelled, no?
+  if (!defined(promise)) {
+    return Promise.resolve(bufferLoader);
```

Comment:
This was one area I wasn't sure about, and could possible be an existing bug. It's possible for `BufferLoader._fetchArrayBuffer` to return `undefined` if the request scheduler is full, meaning the request needs to be retried later. But the existing behavior appeared to swallow the error.

Reply:
`resource.throttle` should be false here so the promise should always be defined. Only the tile's main content (.glb or .gltf) is throttled currently. I edited one of the test tilesets to have an external bin file and confirmed that this is the case.

Reply:
No, that issue is about improving throttling behavior for the things that are being throttled.